### ISSUE TITLE
System notifications for received chat messages while in that chat are not silenced

### ISSUE THREAD

**Issue by e-caste** [nextcloud/spreed-ios#1252]
## Steps to reproduce
1. Open a chat
2. Write back and forth, receiving messages in that chat while having the same chat open (app in foreground)
3. System notifications for new messages of the open chat are still shown, instead of being hidden since the messages are already visible in the chat directly

### Expected behaviour
When you receive messages in a chat, and you have the chat open, the corresponding notifications should not appear.

### Actual behaviour
They do (see steps above).  
I have noticed this behavior in ~50% of cases, not always but it's not even rare.

## Device information

**Device:** iPhone X

**iOS version:** 16.5

**Talk version:** 16.0.2

## Server information

**Nextcloud version:** (see admin overview page: `/index.php/settings/admin/overview`) 26.0.1

**Talk version:** (see apps admin page: `/index.php/settings/apps`) 16.0.4

**Custom Signaling server configured:** yes/no and version (see talk admin settings: `/index.php/settings/admin/talk#signaling_server`) no

**Custom TURN server configured:** yes/no (see talk admin settings: `/index.php/settings/admin/talk#turn_server`) yes

**Custom STUN server configured:** yes/no (see talk admin settings: `/index.php/settings/admin/talk#stun_server`) yes

### Server log (data/nextcloud.log)
<details>

```
Insert your server log here
```
</details>

**Comment by SystemKeeper** [comment]
Hm, this sounds more like a server issue. The problem is as soon as we receive a push notification we _have_ to show that notification as otherwise the app gets killed by the OS. 
Are you seeing the users avatar on that notification or is it the default Talk symbol?

**Comment by e-caste** [comment]
I see, so maybe the server hasn't received the "signal" that the chat is open in the app?  
To answer your question, I see the user's avatar as usual.

**Comment by SystemKeeper** [comment]
I’ll take a look and try to reproduce it. When the server detects that a session is online, no push will be send to prevent exactly what you’re seeing. So could be 2 things, either the „online“ state is not correctly send/received or the server is sending a push when it should not. Not sure yet

**Comment by Ivansss** [comment]
Hi @e-caste 
Could be that those notifications are from reply messages or messages that contain mentions?
If someone reply to one of your messages or mentions you in a message, the server will send a push notification even if you are currently in that room. As @SystemKeeper said, at the moment there is no way to hide that notification from the iOS app, so we need to show it.

**Comment by e-caste** [comment]
@Ivansss I don't think this behavior is limited to replies, I have seen it with simple messages as well. It could be that the first message in a row is a reply and that is what triggers the following notifications, I'll try to notice if this is the case, but if I had to guess this happens too frequently to be caused only by replies.

(I never use mentions so I can't refer to them)
